spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/integration/SqsMessageConversionIntegrationTests.java

```diff
@@ -135,6 +140,17 @@ void resolvesMyOtherPojoFromHeader() throws Exception {
 		assertThat(latchContainer.resolvesMyOtherPojoFromMappingLatch.await(10, TimeUnit.SECONDS)).isTrue();
 	}
 
+	@Test
+	void resolvesMyPojoFromNotificationMessage() throws Exception {
```

Comment:
Here is the test @tomazfernandes. It actually initially failed in the constructor of `NotificationRequestConverter` because the mapper was null.

```java
this.jsonMapper = jsonMapper == null ? new ObjectMapper() :jsonMapper
```

not sure if the reason was because I am using `this.endpointRegistrar.getObjectMapper()` to pass in the objectMapper, and I should be using something else. Or it is just a test issue that I need to address elsewhere. Let me know the best approach

Reply:
We can use a `SqsListenerConfigurer` bean to set the `ObjectMapper` instance.

Reply:
Makes sense?

Reply:
Oh yeah, I did this on line 306, is that the correct place or should I be doing it somewhere else?

Reply:
> Oh yeah, I did this on line 306, is that the correct place or should I be doing it somewhere else?

Yes, in L306 it's correct.

Is it working as expected? Anything else to do regarding this?

Reply:
Yes working as expected, I moved it down in the code where it made a little more sense.

I also added a null check for `objectMapper` in `createAdditionalArgumentResolvers`, I noticed that if objectMapper was null it would cause some issues